hibernate-core/src/main/java/org/hibernate/boot/registry/classloading/internal/ClassLoaderServiceImpl.java

```diff
@@ -157,19 +137,96 @@ private static ClassLoader locateTCCL() {
 		}
 	}
 
+	@Override
+	public TCCLLookupBehavior getTTCLLookupBehavior() {
+		return getAggregatedClassLoader().tcclLookupBehavior;
+	}
+
+	@Override
+	public void setTCCLLookupBehavior(TCCLLookupBehavior behavior) {
+		getAggregatedClassLoader().tcclLookupBehavior = behavior;
+	}
+
 	private static class AggregatedClassLoader extends ClassLoader {
 		private final ClassLoader[] individualClassLoaders;
+		private volatile TCCLLookupBehavior tcclLookupBehavior = TCCLLookupBehavior.AFTER;
```

Comment:
But EntityManagerFactoryBuilderImpl only needs access to set the value on the BootstrapServiceRegistryBuilder - not the ClassLoaderService.  So I agree that this should be made final, and passed into the ClassLoaderServiceImpl ctor.